java/AsyncIO/src/main/java/com/amazonaws/services/msf/RetriesFlinkJob.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.connector.source.util.ratelimit.RateLimiterStrategy;
+import org.apache.flink.connector.datagen.source.DataGeneratorSource;
+import org.apache.flink.connector.kinesis.sink.KinesisStreamsSink;
+import org.apache.flink.streaming.api.datastream.AsyncDataStream;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.LocalStreamEnvironment;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.async.AsyncFunction;
+import org.apache.flink.streaming.api.functions.async.AsyncRetryStrategy;
+import org.apache.flink.streaming.util.retryable.AsyncRetryStrategies;
+import org.apache.flink.streaming.util.retryable.RetryPredicates;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class RetriesFlinkJob {
+        private static final Logger LOGGER = LogManager.getLogger(RetriesFlinkJob.class);
+        private static final String LOCAL_APPLICATION_PROPERTIES_RESOURCE = "flink-application-properties-dev.json";
+
+        private static Map<String, Properties> loadApplicationProperties(StreamExecutionEnvironment env)
+                throws IOException {
+                if (env instanceof LocalStreamEnvironment) {
+                        LOGGER.debug("Loading application properties from '{}'", LOCAL_APPLICATION_PROPERTIES_RESOURCE);
+                        return KinesisAnalyticsRuntime.getApplicationProperties(
+                                RetriesFlinkJob.class.getClassLoader()
+                                        .getResource(LOCAL_APPLICATION_PROPERTIES_RESOURCE).getPath());
+                } else {
+                        LOGGER.debug("Loading application properties from Amazon Managed Service for Apache Flink");
+                        return KinesisAnalyticsRuntime.getApplicationProperties();
+                }
+        }
+
+        public static void main(String[] args) throws Exception {
+                final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+                // Load application parameters
+                final Map<String, Properties> applicationParameters = loadApplicationProperties(env);
+
+                Properties endpointProperties = applicationParameters.getOrDefault("EndpointService", new Properties());
+                String apiUrl = endpointProperties.getProperty("api.url", "");
+                String apiKey = endpointProperties.getProperty("api.key", "");
+
+                Preconditions.checkArgument(!apiUrl.isEmpty(), "API URL must not be empty");
```

Comment:
This is redundant. You are making the same check in the ProcessingFunction constructor